sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SimpleSyncPoller.java

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.implementation.PollContextRequiredException;
+import reactor.core.Exceptions;
+
+import java.time.Duration;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ *
+ * Simple implementation of {@link SyncPoller}.
+ *
+ * @param <T> The type of poll response value
+ * @param <U> The type of the final result of the long running operation
+ */
+public final class SimpleSyncPoller<T, U> implements SyncPoller<T, U> {
```

Comment:
consider renaming this to `DefaultSyncPoller` and moving/renaming existing (private) sync poller to avoid collision.
please get consensus on naming before doing so first.

Other option to pursue is to create SyncPollerFactory (or factory methods on the interface) and not expose concrete implementation at all.